sklearn/preprocessing/_function_transformer.py

```diff
@@ -198,6 +220,63 @@ def inverse_transform(self, X):
             X = check_array(X, accept_sparse=self.accept_sparse)
         return self._transform(X, func=self.inverse_func, kw_args=self.inv_kw_args)
 
+    @available_if(lambda transformer: transformer.feature_names_out is not None)
```

Comment:
I think using `self` is a little clearer here:

```suggestion
    @available_if(lambda self: self.feature_names_out is not None)
```

Reply:
Thanks for the review, @thomasjpfan . You're right, it's clearer with `self`, fixed.